pubs/commands/add_cmd.py

```diff
@@ -86,21 +87,26 @@ def command(conf, args):
         if args.doi is None and args.isbn is None:
             bibentry = bibentry_from_editor(conf, ui, rp)
         else:
+            bibentry = None
```

Comment:
Here the new logic is to try all submitted information in order of DOI, ISBN, Arxiv (eventually overwiting previous results) and then failing if none yield a well formatted result.

With regard to my comment above about using exceptions in the API code, it means that these exceptions should be caught here and yield warnings, not error (there may be a warning about the arxiv not being well formatted although a bibtex could be found through the provided DOI or ISBN). At the end only, an error is printed and the `ui.exit` called if none of the API provided a result.

Currently it could occur that the DOI yield a bibtex that would be overwritten by the Arxiv API returning `None` in the case of https://github.com/pubs/pubs/pull/146/files#diff-a517beca1cb0eb07bc6a2a9ae7e7f544R40. Using an exception as explained above would avoid this and lead to an overall more robust behavior.

We will probably need a [_usecase_](https://github.com/pubs/pubs/blob/master/tests/test_usecase.py) to check this.

Reply:
Yes, good catch.  I've changed it now so that if one of these fails, the program will exit.  I.e., if an invalid DOI is presented along with a valid arXiv ID, the program will still fail.  (Although, if a valid DOI is presented along with an invalid arXiv ID nothing will happen.)  I think this is the least surprising behavior since the user will probably want to know that they have provided invalid information to the program rather than have the program decide what is correct.  But I could be persuaded otherwise.  

I agree that we should have a usecase test to check this, though I haven't added it yet.

Reply:
> I think this is the least surprising behavior since the user will probably want to know that they have provided invalid information

The thing is, right now, the program may fail when a DOI/arXiv/ISBN source reference is valid but not found.

More generally, I see two consistent behaviors we may consider:
 1. accept several of these options (`-X`, `-S`, `-I`) to be set and use one that works (with some precedence rules and show warnings for the others), or open an editor,
 2. accept at most one of these to be set and fail when it is not found.

In particular I don't think it is consistent to require references to be found but then eventually silently ignore the result of the arXiv source because a DOI was set too.

In my opinion option 2. has the least potential ambiguity for the user since it is clear that if a DOI is set it will be used or the program will fail. Also in 1. one could expect that the resulting data (e.g. from a DOI and arXiv) are merged which is not the case. Also, in most cases only one of these options will be used.